### ISSUE TITLE
FTBFS: error: 'MONO_OFFSET_GSharedVtCallInfo_stack_usage' undeclared

### ISSUE THREAD

**Issue by thiagomacieira** [mono/mono#9233]
## Steps to Reproduce

1. Download http://download.mono-project.com/sources/mono/mono-5.12.0.226.tar.bz2
2. Configure as: `--build=x86_64-generic-linux-gnu --host=x86_64-generic-linux-gnu --target=x86_64-clr-linux-gnu --program-prefix= --prefix=/usr --exec-prefix=/usr --bindir=/usr/bin --sbindir=/usr/bin --sysconfdir=/etc --datadir=/usr/share --includedir=/usr/include --libdir=/usr/lib64 --libexecdir=/usr/libexec --localstatedir=/var --sharedstatedir=/usr/com --mandir=/usr/share/man --infodir=/usr/share/info --disable-static`
3. make

### Current Behavior

Mono fails to build from source (FTBFS):
```
../../mono/metadata/abi-details.h:58:43: error: 'MONO_OFFSET_GSharedVtCallInfo_stack_usage' undeclared (first use in this function); did you mean 'MONO_OFFSET_CallContext_stack_size'?
 #define MONO_STRUCT_OFFSET(struct,field) (MONO_OFFSET_ ## struct ## _ ## field == -1, G_STRUCT_OFFSET (struct,field))
                                           ^~~~~~~~~~~~
../../mono/arch/amd64/../x86/x86-codegen.h:274:38: note: in definition of macro 'x86_imm_emit32'
    x86_imm_buf imb; imb.val = (int) (imm); \
                                      ^~~
../../mono/arch/amd64/amd64-codegen.h:346:3: note: in expansion of macro 'amd64_membase_emit'
   amd64_membase_emit ((inst), (reg), (basereg), (disp)); \
   ^~~~~~~~~~~~~~~~~~
../../mono/arch/amd64/amd64-codegen.h:359:3: note: in expansion of macro 'amd64_mov_reg_membase_body'
   amd64_mov_reg_membase_body((inst), (reg), (basereg), (disp), (size)); \
   ^~~~~~~~~~~~~~~~~~~~~~~~~~
tramp-amd64-gsharedvt.c:261:2: note: in expansion of macro 'amd64_mov_reg_membase'
  amd64_mov_reg_membase (code, AMD64_R11, AMD64_RAX, MONO_STRUCT_OFFSET (GSharedVtCallInfo, stack_usage), 4);
  ^~~~~~~~~~~~~~~~~~~~~
tramp-amd64-gsharedvt.c:261:53: note: in expansion of macro 'MONO_STRUCT_OFFSET'
  amd64_mov_reg_membase (code, AMD64_R11, AMD64_RAX, MONO_STRUCT_OFFSET (GSharedVtCallInfo, stack_usage), 4);
                                                     ^~~~~~~~~~~~~~~~~~
../../mono/metadata/abi-details.h:58:43: note: each undeclared identifier is reported only once for each function it appears in
 #define MONO_STRUCT_OFFSET(struct,field) (MONO_OFFSET_ ## struct ## _ ## field == -1, G_STRUCT_OFFSET (struct,field))
                                           ^~~~~~~~~~~~
../../mono/arch/amd64/../x86/x86-codegen.h:274:38: note: in definition of macro 'x86_imm_emit32'
    x86_imm_buf imb; imb.val = (int) (imm); \
                                      ^~~
../../mono/arch/amd64/amd64-codegen.h:346:3: note: in expansion of macro 'amd64_membase_emit'
   amd64_membase_emit ((inst), (reg), (basereg), (disp)); \
   ^~~~~~~~~~~~~~~~~~
../../mono/arch/amd64/amd64-codegen.h:359:3: note: in expansion of macro 'amd64_mov_reg_membase_body'
   amd64_mov_reg_membase_body((inst), (reg), (basereg), (disp), (size)); \
   ^~~~~~~~~~~~~~~~~~~~~~~~~~
tramp-amd64-gsharedvt.c:261:2: note: in expansion of macro 'amd64_mov_reg_membase'
  amd64_mov_reg_membase (code, AMD64_R11, AMD64_RAX, MONO_STRUCT_OFFSET (GSharedVtCallInfo, stack_usage), 4);
  ^~~~~~~~~~~~~~~~~~~~~
tramp-amd64-gsharedvt.c:261:53: note: in expansion of macro 'MONO_STRUCT_OFFSET'
  amd64_mov_reg_membase (code, AMD64_R11, AMD64_RAX, MONO_STRUCT_OFFSET (GSharedVtCallInfo, stack_usage), 4);
                                                     ^~~~~~~~~~~~~~~~~~
```

File `./mono/metadata/object-offsets.h` has:
```
#if defined(TARGET_ARM)
DECL_OFFSET(MonoLMF, method)
DECL_OFFSET(GSharedVtCallInfo, stack_usage)
....
#endif

#if defined(TARGET_ARM64)
DECL_OFFSET (MonoContext, has_fregs)

DECL_OFFSET(GSharedVtCallInfo, stack_usage)
....
#endif
```

All other uses of the pair `GSharedVtCallInfo` and `stack_usage` together in the same line are the lines causing the build errors.

### Expected Behavior

Build succeeds.

## On which platforms did you notice this

[ ] macOS
[x] Linux
[ ] Windows

**Version Used**: 5.12.0

**Comment by thiagomacieira** [comment]
The `--build=x86_64-generic-linux-gnu --host=x86_64-generic-linux-gnu --target=x86_64-clr-linux-gnu` argument set causes the build failure.  If the three arguments are identical, it works.

**Comment by akoeplinger** [comment]
@thiagomacieira why are you building with `--target=x86_64-clr-linux-gnu` ?

**Comment by thiagomacieira** [comment]
The %configure macro in the .spec file does that. It's worked for everything else so far.

**Comment by akoeplinger** [comment]
which .spec file are you talking about?

**Comment by luhenry** [comment]
@alexanderkyte could you please take a look at that?

**Comment by thiagomacieira** [comment]
> which .spec file are you talking about?

The one I'm trying to write. The RPM .spec file was:
```
%configure --disable-static
```

The other arguments are automatic from the distro.

I'm guessing the buildsystem concluded it was a cross-compilation but something is not right.

**Comment by akoeplinger** [comment]
ohh, this is ClearLinux right? I was getting confused by the `clr` in the `target` triplet thinking there was some special support for the CLR (Common Language Runtime)... 😄

**Comment by thiagomacieira** [comment]
Right, Clear Linux. Sorry for the confusion.

By the way, I had to disable `make check` as that would fail. Is that expected? Are you interested in the logs?

**Comment by akoeplinger** [comment]
If it's a test that fails consistently then creating a new issue would be nice.

I wouldn't run `make check` as part of packaging, we're running a few hundred thousand tests and something somewhere always fails ;)
